app/src/main/java/tech/zafrani/companionforpubg/fragments/PUBGMapFragment.java

```diff
@@ -29,65 +29,37 @@
     @Nullable
     private GoogleMapControllerImpl mapController = null;
 
-    private ImageView vehicleIcon;
+    @BindView(R.id.fragment_map_vehicle_icon)
+    ImageView vehicleIcon;
 
-    private ImageView boatIcon;
+    @BindView(R.id.fragment_map_boat_icon)
+    ImageView boatIcon;
 
-    private ImageView runDistanceIcon;
+    @BindView(R.id.fragment_map_distance_icon)
+    ImageView runDistanceIcon;
 
     @Nullable
     private SharedPreferences sharedPreferences = null;
 
     //region BaseFragment
-    @Nullable
     @Override
-    public View onCreateView(final LayoutInflater inflater,
-                             @Nullable final ViewGroup container,
-                             final Bundle savedInstanceState) {
-        return inflater.inflate(R.layout.fragment_map, container, false);
+    protected int getLayoutRes() {
+        return R.layout.fragment_map;
     }
 
     @Override
     public void onViewCreated(final View view,
                               @Nullable final Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
         this.sharedPreferences = getActivity().getSharedPreferences("temp", Context.MODE_PRIVATE); //todo not this.
-        this.vehicleIcon = (ImageView) view.findViewById(R.id.fragment_map_vehicle_icon);
-        this.boatIcon = (ImageView) view.findViewById(R.id.fragment_map_boat_icon);
-        this.runDistanceIcon = (ImageView) view.findViewById(R.id.fragment_map_distance_icon);
 
         final MapFragment mapFragment = (MapFragment) getChildFragmentManager().findFragmentById(R.id.map_fragment_map);
         mapFragment.getMapAsync(this);
-
-        this.vehicleIcon.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                toggleVehicles();
-            }
-        });
-        this.boatIcon.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                toggleBoats();
-            }
-        });
-        this.runDistanceIcon.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                toggleRunDistance();
-            }
-        });
     }
 
     @Override
     public void onDestroy() {
         super.onDestroy();
-        if (this.vehicleIcon != null) {
-            this.vehicleIcon.setOnClickListener(null);
-        }
-        if (this.runDistanceIcon != null) {
-            this.runDistanceIcon.setOnClickListener(null);
```

Comment:
I'm not 100% sure if butterknife removes the listeners so I'm going to add and remove the listeners manually